### ISSUE TITLE
Parallel file conversion

### ISSUE THREAD

**Issue by leewujung** [OSOceanAcoustics/echopype#217]
Obviously It is useful to be able to convert multiple files in parallel. The new Convert structure makes it more straightforward to enable this. 

This was originally mentioned in #181. Making a separate issue for easier tracking.

**Comment by leewujung** [comment]
Seems better to close this and start a new one when we're ready go take this on, with more up-to-date description on code sections to be changed to enable this. We've decided that the current priority is to be able to deal with problems in converting files that can cause memory exploration either during the dataarray merging process or due to NaN padding.
